### PR TITLE
CLOUDP-93399: Add support for Cloud Gov

### PR DESCRIPTION
**Pull request by andreaangiolillo**
Add support for Cloud Gov

Evergreen Task: [atlas_gov_generic_e2e](https://evergreen.mongodb.com/task/mongocli_master_e2e_generic_atlas_gov_generic_e2e_patch_fd84d0e3309a5248828f72587fc1026f4ae21612_60e434660305b96bf1bb8361_21_07_06_10_45_59)

## Proposed changes

_Jira ticket:_ CLOUDP-93399

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli config --service=cloudgov
You are configuring a profile for mongocli.

All values are optional and you can use environment variables (MCLI_*) instead.

Enter [?] on any option to get help.

? Public API Key: ddvzdkiw
? Private API Key: [? for help] ************************************
? Choose a default organization: Test CLI (60e409d39e80a53e4f4163a7)
? Choose a default project: AndreaCLI (60e40e9e9e80a53e4f41686e)
? Default Output Format: json
? Default MongoDB Shell Path: /usr/local/bin/mongosh

Your profile is now configured.
You can use [mongocli config set] to change these settings at a later time.
```
**mongocli.toml:**
```
mongosh_path = "/usr/local/bin/mongosh"

[default]
  org_id = "60e409d39e80a53e4f4163a7"
  output = "json"
  private_api_key = "03dcc7ec-bf24-43d8-9d14-2c061592077f"
  project_id = "60e40e9e9e80a53e4f41686e"
  public_api_key = "ddvzdkiw"
  service = "cloudgov"
```

### CODE REVIEW

File: internal/config/const.go
Comment (andreaangiolillo) [inline internal/config/const.go:41]: I will update this link before merging the PR

**Comment by andreaangiolillo** [comment]
@gssbzn Could you have a look at this PR and tell me if there is anything that I should change, please? 
I will start implementing the e2e tests once I have your initial approval. Thanks!

File: internal/cli/config/config.go
Comment (gssbzn) [inline internal/cli/config/config.go:60]: this may work for now, but not all services are available to gov, data lake is one of them but we can refine this later

File: internal/store/store.go
Comment (gssbzn) [inline internal/store/store.go:283]: [q] does this need to be exported?

File: internal/cli/config/config.go
Comment (andreaangiolillo) [reply internal/cli/config/config.go:60]: if the service is not supported in gov, couldn’t we just not add `config.CloudGovService` to the function in the store?

Comment (andreaangiolillo) [reply internal/cli/config/config.go:60]: I mean in the switch inside the store file:
```
func (s *Store) Projects(opts *atlas.ListOptions) (interface{}, error) {
	switch s.service {
	case config.CloudService, config.CloudGovService:
		result, _, err := s.client.(*atlas.Client).Projects.GetAllProjects(context.Background(), opts)
```

If a service is not implemented for gov we could just remove `config.CloudGovService ` from the related function in the store.

**Review by gssbzn — COMMENTED** [review]
the `IsCloud` may need future work and refieentment but for now this looks like a solid approach 👍

File: internal/cli/config/config.go
Comment (gssbzn) [reply internal/cli/config/config.go:60]: yep, something like that may be the way forward we just need to verify what's available and what not and that may take some time

Comment (gssbzn) [reply internal/cli/config/config.go:60]: resolving this for now

File: internal/store/projects.go
Comment (gssbzn) [inline internal/store/projects.go:197]: [q] what's the plan here, do you plan to test command by command to enable api access? it may be easier to enable for all and then start restricting but want to hear your thoughts

**Review by gssbzn — APPROVED** [review]
great stuff, lgtm with a question about the store methods not enabled yet

File: internal/store/projects.go
Comment (andreaangiolillo) [reply internal/store/projects.go:197]: I was thinking about creating another ticket where I test and enable all the commands for cloud-gov. Yes, I would enable  for all in my local machine (except the ones that we are sure are not there -> datalakes etc) and see if they work but add in my PR only the ones that I am sure are working

**Pull request merged**
